### DISCUSSION TITLE
General concept of batching

### DISCUSSION THREAD

**Discussion by mkoohafkan** [ropensci/targets#734] [General]
The documentation on [dynamic branching](https://books.ropensci.org/targets/dynamic.html#dynamic-branching-performance) cautions that large numbers of targets can produce significant overhead, and goes on to describe [batching](https://books.ropensci.org/targets/dynamic.html#batching)  as a way to improve performance.

Is batching a way to "chunk out" a workflow so that targets does not try to build or interpret the whole network/workflow at once? Or does batching simply to reduce the total number of targets, e.g., by merging multiple targets into a single "batch" (so that the whole  batch needs to be rerun if something is modified)?

**Comment by wlandau** [comment]
> Is batching a way to "chunk out" a workflow so that targets does not try to build or interpret the whole network/workflow at once?

Sorry, I am not sure what you mean.

> Or does batching simply to reduce the total number of targets, e.g., by merging multiple targets into a single "batch" (so that the whole batch needs to be rerun if something is modified)?

Exactly, it reduces the total number of targets so there is less overhead. Optimizing batch size is a balance between that overhead versus the cost and risk of invalidating a batch.

**Reply by mkoohafkan** [reply]
thanks, you answered my question. By "chunking" I was asking if batching was used to reduce overhead by splitting up the evaluation or checking of targets into groups to reduce overhead but still explicitly maintaining the various targets within a batch.

I'm visualizing batching as a graph contraction (multiple nodes are merged into one) whereas by "chunking" I was thinking the graph was split into multiple subgraphs (which may or may not share nodes) so that each subgraph could be evaluated separately.

**Reply by wlandau** [reply]
In batching, each batch is a target (a dynamic branch) that contains the output of multiple replications (or function calls) of the original task. More work is nested inside individual targets, as opposed to nesting targets inside one another or inside abstract chunks.
